File: 7.cpp
#include <iostream>
#include <iomanip>

int main() {

	int num, i= 0, j;
	bool flag = true;

	std::cout << "*********************************"
		<< "      MULTIPLICATION TABLES      "
		<< "*********************************"
		<< std::endl << std::endl;

	do {
		std::cout << std::endl << std::endl
			<< "ENTER A NUMBER FROM 1 TO 12 (0 TO EXIT): ";
		std::cin >> num;
		if (std::cin.fail() || num < 0 || num > 12) {
			std::cin.clear();
			std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
		}
		else {
			flag = false;
		}
	} while (flag == true);

	if (num == 0) {
		std::cout << std::endl << std::endl
			<< "GOODBYE! :)"
			<< std::endl << std::endl;
		exit(0);
	}

	std::cout << std::endl << std::endl
		<< " ---------------"
		<< std::endl;
	while (i != num) {
		j = i * num;
		std::cout << "| " << std::left << std::setw(2) << i << " X " << num << " = "
			<< std::setw(3) << j << " |"
			<< std::endl;
		i++;
	}
	j = i * num;
	std::cout << "| " << std::left << std::setw(2) << i << " X " << num << " = "
		<< std::setw(3) << j << " |"
		<< std::endl;
	std::cout << " ---------------"
		<< std::endl;

	return 0;
}

File: 7.py
print("\n\n\n*********************************")
print("      MULTIPLICATION TABLES      ")
print("*********************************")
print("\n\n___________________________________________________________________________________________________\n\n")

while True:
    data = input("\n\nENTER A NUMBER FROM 1 TO 12 (0 TO EXIT): ")
    checkData = data.strip()
    if checkData.isdigit():
        num = int(checkData)
        if num <= 12 and num >= 1:
            break
        elif num == 0: exit(0)

print("\n\n ---------------")

i = 0

while i != num:
    j = i * num
    print("| {:<2d} X {:<2d} = {:<3d} |".format(i, num, j))
    i += 1

j = i * num
print("| {:<2d} X {:<2d} = {:<3d} |".format(i, num, j))
print(" ---------------\n\n\n")


'''
    <	Left aligned to the remaining space
    ^	Center aligned to the remaining space
    >	Right aligned to the remaining space
    =	Forces the signed (+) (-) to the leftmost position
'''
